Sources/UITextFieldExtensions.swift

```diff
@@ -47,3 +47,43 @@ extension UITextField {
     }
 
 }
+
+
+/// MARK: Validations
+extension UITextField {
+    /// EZSE: Ways to validate by comparison
+    enum textFieldValidationOptions: Int {
+        case equalTo, greaterThan, greaterThanOrEqualTo, lessThan, lessThanOrEqualTo
+    }
+    
+    /// EZSE: Validation length of character counts in UITextField
+    func validateLength(ofCount count: Int, option: UITextField.textFieldValidationOptions) -> Bool {
+        switch option {
+        case .equalTo:
+            return self.text!.characters.count == count
+        case .greaterThan:
+            return self.text!.characters.count > count
+        case .greaterThanOrEqualTo:
+            return self.text!.characters.count >= count
+        case .lessThan:
+            return self.text!.characters.count < count
+        case .lessThanOrEqualTo:
+            return self.text!.characters.count <= count
+        }
+    }
+
+    /// EZSE: Validation of email format based on https://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address and https://stackoverflow.com/questions/2049502/what-characters-are-allowed-in-an-email-address
+    // TODO match String.isEmail method
+    func validateEmail() -> Bool {
+        let emailRegEx = "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"
```

Comment:
No it does not have to be in string class. It can be a member in UITextField class (so just put it as a member in this extension).